### PR TITLE
enable dhcp as optional parameter (typelist)

### PR DESCRIPTION
**Pull request by MalloZup**
This pr contain: 

- dhcp backport from #288.
- documentation.

Additionaly to the #288 we have:
-  real testing for real the dhcp parameter via xml property of network
    -  we are testing that the xml networkDefinition have dhcp set or not (depending on dhcp nabl/disable)

### CODE REVIEW

**Comment by MalloZup** [comment]
https://github.com/dmacvicar/terraform-provider-libvirt/issues/389

**Comment by MalloZup** [comment]
@dmacvicar feel free to review once you have free time. tests are passing and is done.

**Comment by Bischoff** [comment]
@MalloZup just pick the fixes to the crash mentioned above from my own PR:
* https://github.com/dmacvicar/terraform-provider-libvirt/pull/321/files#diff-f6f8db75203b0db83110f024f21cc4eb#232 `if ips.DHCP != nil ` ...
* https://github.com/dmacvicar/terraform-provider-libvirt/pull/321/files#diff-042e2374df248e4277a3506b065eb28a#243 `if enableDHCP {` ...

I think it will not require much adaptation.

**Comment by MalloZup** [comment]
Thx for feedback.i will have a look once i have a  terminal.enojoy your weekend

File: libvirt/resource_libvirt_network_test.go [hunk 1]
```diff
@@ -130,3 +133,96 @@ func TestAccLibvirtNetwork_Import(t *testing.T) {
 		},
 	})
 }
+
+func TestAccLibvirtNetwork_EnableDhcpAndDisableItAfter(t *testing.T) {
```
Comment (MalloZup) [inline libvirt/resource_libvirt_network_test.go]: rename this

Comment (Bischoff) [reply libvirt/resource_libvirt_network_test.go]: To whom are you speaking to? :smile_cat:

Comment (MalloZup) [reply libvirt/resource_libvirt_network_test.go]: i just reviewed my code myself :rocket: :robot: was a sidenote :rofl:

Comment (Bischoff) [reply libvirt/resource_libvirt_network_test.go]: Please don't get angry with your reviewer :yum: . Enjoy your week-end too :fireworks: .

**Comment by MalloZup** [comment]
@Bischoff  in reply to your 2nd comment:

- https://github.com/dmacvicar/terraform-provider-libvirt/pull/321/files#diff-042e2374df248e4277a3506b065eb28a#243 if enableDHCP { ...

This is already here:

https://github.com/dmacvicar/terraform-provider-libvirt/pull/385/files#diff-042e2374df248e4277a3506b065eb28aR458

if the dhcp is not enabled we set it to nil which is equivalent to your suggestion.

For the domain.go yop i added the check :+1:  Documentation is also improved/

@dmacvicar  it is interesting the golint/go vet didn"t catch the "not formatted" function variable, i think it can be maybe an issue upstream.

Feel free to review and test it. TIA

**Comment by Bischoff** [comment]
@MalloZup ok for the second comment already applied.

For the first comment, I applied it locally, and terraform does not crash anymore.

**Comment by MalloZup** [comment]
we could do the `dns = enablement` in separate PR and track this in an ISSUE

**Pull request merged**